07.bowling_object/bowling_object.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+def main
+  score = ARGV[0]
+  scores = score.split(',')
+  shots = []
+  scores.each do |s|
+    if s == 'X'
+      if shots.size < 18
+        shots << 'X'
+        shots << 0
+      else
+        shots << 'X'
+      end
+    else
+      shots << s.to_i
+    end
+  end
+  game = Game.new(shots)
+  p game.score
+end
+
+class Game
+  attr_reader :frames
+  def initialize(frames)
+    @sum = 0
+    (0...20).each_slice(2) do |a, b|
+      if a < 18
+        frame = Frame.new(frames[a], frames[b])
+        @sum +=  if frame.judgment == 'spare'
+                  Shot.new(frames[a + 2]).score
+                elsif frame.judgment == 'strike'
+                  if a < 16
+                    if Shot.new(frames[a + 2]).score == 10
+                      Shot.new(frames[a + 2]).score + Shot.new(frames[a + 4]).score
+                    else
+                      Frame.new(frames[a + 2],frames[b + 2]).score
+                    end
+                  else
+                    Frame.new(frames[a + 2],frames[b + 2]).score
+                  end
+                end
+      else
+        frame = Frame.new(frames[a], frames[b], frames[a + 2])
+      end
+      @sum += frame.score
+    end
+  end
+  def score
+    @sum
+  end
+end
+
+class Frame
+  attr_reader :first_shot, :second_shot, :third_shot
+  def initialize(first_mark, second_mark, third_mark = nil)
+    @first_shot = Shot.new(first_mark)
+    @second_shot = Shot.new(second_mark)
+    @third_shot = Shot.new(third_mark)
+  end
+  def score
+    [@first_shot.score, @second_shot.score, @third_shot.score].sum
+  end
+  def judgment
+    if @first_shot.score == 10
+      return 'strike'
+    elsif [@first_shot.score, @second_shot.score].sum == 10
+      return 'spare'
+    end
+  end
+  def first_shot_score
+    @first_shot.score
+  end
+  def second_shot_score
+    @second_shot.score
+  end
+end
+
+class Shot
+  attr_reader :mark
+  def initialize(mark)
+    @mark = mark
+  end
+  def score
+    return 10 if mark == 'X'
+    mark.to_i
+  end
+end
```

Comment:
書き方の指摘になりますが、メソッドとの間には1行スペース入れたほうが見やすいですね。
```suggestion
class Shot
  attr_reader :mark
  
  def initialize(mark)
    @mark = mark
  end
  
  def score
    return 10 if mark == 'X'
    mark.to_i
  end
end
```